samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -311,6 +321,68 @@ private StreamPartitionCountMonitor getPartitionCountMonitor(Config config, Syst
       });
   }
 
+  private Optional<InputStreamRegexMonitor> getInputRegexMonitor(Config config, SystemAdmins systemAdmins) {
+
+    // if input regex monitor is not enabled return empty
+    if (new JobConfig(config).getMonitorInputRegexFrequency() <= 0) {
+      log.info("InputStreamRegexMonitor is disabled.");
+      return Optional.empty();
+    }
+
+    StreamMetadataCache streamMetadata = new StreamMetadataCache(systemAdmins, 0, SystemClock.instance());
+    Set<SystemStream> inputStreamsToMonitor = new TaskConfigJava(config).getAllInputStreams();
```

Comment:
Why do we need this Set? IIUC, the monitor should be able to auto-construct it during startup since it's already watching a particular regex.

Reply:
race condition w/ new topic additions